src/main/java/com/williamfiset/algorithms/strings/BoyerMooreStringSearch.java

```diff
@@ -21,11 +22,7 @@
       return new ArrayList<>();
     }
     List<Integer> occurrences = new ArrayList<>();
-
-    Map<Character, Integer> skipTable = new HashMap<>();
-    for (int i = 0; i < pattern.length(); i++) {
-      skipTable.put(pattern.charAt(i), pattern.length() - i - 1);
-    }
+    Map<Character, TreeSet<Integer>> skipTable = generateSkipTable(pattern);
```

Comment:
I think using a Map + a TreeMap is huge overkill for this algorithm that will cause lots of memory overhead and slowness. I had a look at 3+ implementations, and none of the used a data structure like this.

Please refactor and have a look at some examples:
1. http://www.personal.kent.edu/~rmuhamma/Algorithms/MyAlgorithms/StringMatch/boyerMoore.htm
2. https://www.geeksforgeeks.org/boyer-moore-algorithm-for-pattern-searching/

Reply:
I think this can still use some improvements. I see a lot of potential to improve the runtime with very simple optimizations. I can handle this myself later, the important thing is that the algorithm is now fixed.

Reply:
Can you point a few of those improvements, curious to know!!!